src/en/guide/basics/routing.md

```diff
@@ -277,7 +280,43 @@ async def handler(request, foo: str):
 This gives you the freedom to define specific matching patterns for your use case. 
 
 In the example shown, we are looking for a date that is in `YYYY-MM-DD` format.
+
+::: tip Advanced
+
+But for complex routing, the above example is still too simple.
+
+Sometimes you want to match on a part of a path segment:
+
+```text
+/image/123456789.jpg
+```
+
+If you wanted to match the file pattern, but only capture the numeric portion, you need to do some regex fun:
+
+```python
+app.route(r"/image/<img_id:(?P<img_id>\d+)\.jpg>")
```

Comment:
I think we should offer this as the standard pattern and not scare anyone away with named matching if we do not need to:

```python
app.get(r"/<foo:(\d+).jpg>")
```